docs/specs/workflow-customization.md

```diff
@@ -0,0 +1,1486 @@
+- We want to keep the guarantee that the existing configuration ensures every new user has some certain identities and authenticators.
+- We want to be able to fulfill the authentication flows in existing consumer apps
+
+### What is a signup flow
+- Create 1 or more Identities. Later on, the User identify themselves with one of the Identities.
+- Create 0 or more Authenticators. The User authenticates themselves with one of the Authenticators if needed.
+- (Optional) Collect user profile (i.e. standard attributes and custom attributes)
+
+### What is a login flow
+- Identify the User with an Identity
+- Depending on the Identity, authenticator the User with the Authenticators. Note that the pre-selected Authenticator is usually associated with the Identity.
+- (Optional) Further authenticate the User with **other** Authenticators.
+
+Suppose the User has a Email Login ID Identity `johndoe@gmail.com`, a Email OOB-OTP Authenticator `johndoe@gmail.com`, a Phone Login ID Identity `+85298765432`, a Phone OOB-OTP Authenticator `+85298765432`, a Password Authenticator, and a OAuth Identity `johndoe@gmail.com`.
+
+If the User identifies themselves with the Email Login ID Identity `johndoe@gmail.com`, then the User can authenticate themselves with:
+1. Email OOB-OTP Authenticator `johndoe@gmail.com`
+2. Password Authenticator
+3. Phone OOB-OTP Authenticator `+85298765432`. Note that this Authenticator is NOT associated with the identifying Identity, but it can also be used to authenticate the User.
+
+If the User identifies themselves with the OAuth Identity `johndoe@gmail.com`, then the User DOES NOT need to authenticate themselves. This is how most other applications work.
+
+### What is a reauth flow
+- Authenticate the User with any Authenticators.
+
+### Design of the configuration
+
+#### Design overview
+
+The configuration is a small DSL with a simple expression language.
+Every object must have an `id`. The namespace in which the `id` is in varies by object.
+
+The configuration maps to a user flow as performed by a end-user.
+A user flow consists of one or more screens, and the configuration consists of one or more steps.
+Screens are not nested, so steps are organized linearly, and not nested as well.
+
+#### Object: IdentificationMethod
+
+We define available IdentificationMethods under `identification_methods`.
+The `id` is in the global namespace.
+
+Example:
+
+```yaml
+identification_methods:
+# Identify the User by a Email Login ID Identity
+- id: email
+  type: "login_id"
+  login_id:
+    type: "email"
+# Identify the User by a Phone Login ID Identity
+- id: phone
+  type: "login_id"
+  login_id:
+    type: "phone"
+# Identify the User by a Username Login ID Identity
+- id: username
+  type: "login_id"
+  login_id:
+    type: "username"
+# Identify the User by a OAuth Identity
+- id: oauth
+  type: "oauth"
+  oauth:
+    aliases: ["google", "apple"]
+# Identify the User by a Anonymous Identity
+- id: anonymous
```

Comment:
Is it really technically possible to have any combination of `anonymous`, `biometric` flow? (Passkey maybe...?)

If not, maybe we shouldn't put them into spec and allow configuration?